File: JanperCab/5-JanperCab/ClientApp/src/app/_models/machine/MachineProductionListDto.ts
import { Type } from 'class-transformer';
import { MachineTypeEnum } from 'src/app/_enums/MachineTypeEnum';
import { MachineProdutionCurrentProcessDto } from './MachineProdutionCurrentProcessDto';

export class MachineProductionListDto {
  id: number;
  machineType: MachineTypeEnum;
  name: string;

  @Type(() => MachineProdutionCurrentProcessDto)
  currentProcesses: MachineProdutionCurrentProcessDto[];
}

File: JanperCab/3-Application/Dtos/Machine/MachineProductionListDto.cs
using _1_Domain.Enum;
using System.Collections.Generic;

namespace _3_Application.Dtos.Machine
{
    public class MachineProductionListDto
    {
        public int Id { get; set; }

        public MachineTypeEnum MachineType { get; set; }

        public string Name { get; set; }

        public List<MachineProdutionCurrentProcessDto> CurrentProcesses { get; set; }
    }
}